src/adapters/CCIPAdapter.sol

```diff
@@ -0,0 +1,122 @@
+        destinations[centrifugeId] = CCIPDestination(chainSelector, adapter);
+        emit Wire(centrifugeId, chainSelector, adapter);
+    }
+
+    /// @inheritdoc IAdapterWiring
+    function isWired(uint16 centrifugeId) external view returns (bool) {
+        return destinations[centrifugeId].chainSelector != 0;
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Incoming
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IAny2EVMMessageReceiver
+    function ccipReceive(IClient.Any2EVMMessage calldata message) external {
+        require(msg.sender == address(ccipRouter), InvalidRouter());
+
+        CCIPSource memory source = sources[message.sourceChainSelector];
+        require(source.addressHash != bytes32(""), InvalidSourceChain());
+
+        address sourceAddress = abi.decode(message.sender, (address));
+        require(source.addressHash == keccak256(abi.encodePacked(sourceAddress)), InvalidSourceAddress());
+
+        entrypoint.handle(source.centrifugeId, message.data);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Outgoing
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IAdapter
+    function send(uint16 centrifugeId, bytes calldata payload, uint256 gasLimit, address)
+        external
+        payable
+        returns (bytes32 adapterData)
+    {
+        require(msg.sender == address(entrypoint), NotEntrypoint());
+        CCIPDestination memory destination = destinations[centrifugeId];
+        require(destination.chainSelector != 0, UnknownChainId());
+
+        IClient.EVM2AnyMessage memory ccipMessage = IClient.EVM2AnyMessage({
+            receiver: abi.encode(destination.addr),
+            data: payload,
+            tokenAmounts: new IClient.EVMTokenAmount[](0),
+            feeToken: address(0), // Pay in native token
+            extraArgs: _argsToBytes(IClient.EVMExtraArgsV1({gasLimit: gasLimit + RECEIVE_COST}))
+        });
+
+        adapterData = ccipRouter.ccipSend{value: msg.value}(destination.chainSelector, ccipMessage);
+    }
+
+    /// @inheritdoc IAdapter
+    function estimate(uint16 centrifugeId, bytes calldata payload, uint256 gasLimit) external view returns (uint256) {
+        CCIPDestination memory destination = destinations[centrifugeId];
+        require(destination.chainSelector != 0, UnknownChainId());
+
+        IClient.EVM2AnyMessage memory ccipMessage = IClient.EVM2AnyMessage({
+            receiver: abi.encode(destination.addr),
+            data: payload,
+            tokenAmounts: new IClient.EVMTokenAmount[](0),
+            feeToken: address(0),
+            extraArgs: _argsToBytes(IClient.EVMExtraArgsV1({gasLimit: gasLimit + RECEIVE_COST}))
+        });
```

Comment:
NIT. Could we move this to some private function to ensure we use the same in `send` & `estimate`?

Reply:
Good idea!